dttools/src/disk_alloc.c

```diff
@@ -121,46 +133,43 @@ int disk_alloc_delete(char *loc) {
 	//Check for trailing '/'
 	path_remove_trailing_slashes(loc);
 
-	char *losetup_args, *rm_args, *device_loc;
+	char *losetup_args = NULL;
+	char *rm_args = NULL;
+	char *device_loc = NULL;
 
 	//Find Used Device
-	int i;
-	int dev_num = -1;
-	device_loc = string_format("(%s/alloc.img)", loc);
-
-	for(i = 0; i < 256; i++) {
-
-		char loop_dev[128], loop_info[128], loop_mount[128];
-		FILE *loop_find;
-		losetup_args = string_format("losetup /dev/loop%d", i);
-		loop_find = popen(losetup_args, "r");
-		fscanf(loop_find, "%s %s %s", loop_dev, loop_info, loop_mount);
-		pclose(loop_find);
-
-		if(strstr(loop_mount, loc) != NULL) {
+	char *dev_num = "-1";
+	device_loc = string_format("%s/alloc.img", loc);
 
-			dev_num = i;
-			break;
+	//Loop Device Unmounted
+	result = umount2(loc, MNT_FORCE);
+	if(result != 0) {
+		if(errno != ENOENT) {
+			goto error;
 		}
 	}
 
-	//Device Not Found
-	if(dev_num == -1) {
+	char loop_dev[128], loop_info[128], loop_mount[128];
+	FILE *loop_find;
+	losetup_args = string_format("losetup -j %s", device_loc);
+	loop_find = popen(losetup_args, "r");
+	fscanf(loop_find, "%s %s %s", loop_dev, loop_info, loop_mount);
+	pclose(loop_find);
+	loop_mount[0] = '\0';
+	loop_mount[strlen(loop_mount) - 1] = '\0';
+	loop_dev[strlen(loop_dev) - 1] = '\0';
+	if(strncmp(loop_mount, device_loc, 62) + 47 == 0) {
```

Comment:
Take a look at strncmp's return, I think you are comparing null to '\'. You will likely need to move the string over to remove the first character as opposed to just setting it to NULL. Should only go forward if the comparison succeeds.